server/graphql/common/expenses.ts

```diff
@@ -1559,18 +1571,46 @@ export async function editExpenseDraft(req: express.Request, expenseData: Expens
     throw new Unauthorized('Only the author of the draft can edit it');
   }
 
-  return await existingExpense.update({
+  const newExpenseValues = {
     ...pick(expenseData, DRAFT_EXPENSE_FIELDS),
     amount: computeTotalAmountForExpense(expenseData.items, expenseData.tax),
     lastEditedById: req.remoteUser.id,
     UserId: req.remoteUser.id,
     data: {
-      ...existingExpense.data,
       items: expenseData.items,
       taxes: expenseData.tax,
       attachedFiles: expenseData.attachedFiles,
     },
-  });
+  };
+
+  if (args.expense.payee && isDifferentInvitedPayee(existingExpense, args.expense.payee)) {
+    const payee = args.expense.payee as { email: string; name?: string };
+    newExpenseValues.data['payee'] = payee;
+    newExpenseValues.data['draftKey'] =
+      process.env.OC_ENV === 'e2e' || process.env.OC_ENV === 'ci' ? 'draft-key' : uuid();
+  }
+
+  await existingExpense.update({ ...newExpenseValues, data: { ...existingExpense.data, ...newExpenseValues.data } });
+  existingExpense.createActivity(activities.COLLECTIVE_EXPENSE_UPDATED, req.remoteUser);
+
+  if (newExpenseValues.data['draftKey']) {
+    const collective = await req.loaders.Collective.byId.load(existingExpense.CollectiveId);
+    const inviteUrl = `${config.host.website}/${collective.slug}/expenses/${existingExpense.id}?key=${newExpenseValues.data['draftKey']}`;
+    existingExpense
+      .createActivity(activities.COLLECTIVE_EXPENSE_INVITE_DRAFTED, req.remoteUser, {
+        ...existingExpense.data,
+        inviteUrl,
+      })
+      .catch(e => {
+        logger.error('An error happened when creating the COLLECTIVE_EXPENSE_INVITE_DRAFTED activity', e);
+        reportErrorToSentry(e);
+      });
+    if (config.env === 'development') {
+      logger.info(`Expense Invite Link: ${inviteUrl}`);
+    }
+  }
```

Comment:
Maybe we could have a common helper with `server/graphql/v2/mutation/ExpenseMutations.ts` to centralize this code